### PR TITLE
[fix](mtmv)fix nested mtmv not refresh

### PR DESCRIPTION
**Pull request by zddr**
fix nested mtmv not refresh because the partition version remains unchanged after inserting overwrite for the underlying materialized view

we add partitionId in snapshot

### CODE REVIEW

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run cloud_p0

File: fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVRefreshPartitionSnapshot.java [hunk 1]
```diff
@@ -74,6 +77,46 @@ public String toString() {
     }
 
     public void compatible(MTMV mtmv) {
+        try {
+            // snapshot add partitionId resolve problem of insert overwrite
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVRefreshPartitionSnapshot.java:81]: add upgrade downgrade test case

**Comment by zddr** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by seawinde — APPROVED** [review]

**Pull request merged**